externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -121,6 +139,25 @@ def Tcp_MulOp : Tcp_BinaryElementwiseOp<"mul", [SameOperandsAndResultElementType
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_DivOp : Tcp_BinaryElementwiseOp<"div", [SameOperandsAndResultElementType]> {
+  let summary = "Computes elementwise division";
```

Comment:
You probably need two division operations: signed & unsigned.

Reply:
That's a good point. Do we have a preference between separate ops for float, signed int, unsigned int, versus using a `bool` attribute indicating signed/unsigned if integer type?

Reply:
My preference is to have three variants: `fdiv`, `sidiv` and `uidiv`.  Unclear how quantization fits in here though.

Reply:
With real (not-fake) quantization, tensors from a quantized op will be no different than regular integers, so IIUC this construct would still hold. LMK if I'm missing something (cc: @sjarus ). 

Also, hyper minor point on the naming for multiple ops - we may want to use `divf`, `divsi`, and `divui` to keep the same prefix across all div (like in `arith` dialect) (cc: @navahgar ).

Reply:
I was thinking of a `bool` attribute, only because it will keep the opset smaller. But thats not a strong preference though.

If we are going with multiple ops, `divf`, `divsi`, and `divui` sounds good to me.

Reply:
Noticed that using separate ops allows easier definition of type constraints for inputs and outputs (e.g. `Tcp_FloatTensor` or `Tcp_IntTensor`), whereas in the combined op we'd use `Tcp_FloatOrIntTensor` type, requiring a separate verifier to do the type legalization.

Reply:
Not sure I see why we need separate verifier even with `Tcp_FloatOrIntTensor` type. 

Are you suggesting that to verify that both inputs have the same element type? We could use `SameOperandsElementType` trait for that.

Reply:
Error when a `tcp.divf` op receives an `i32` operand (without a separate verifier):
```
unexpected error: 'tcp.divf' op operand #0 must be ranked tensor of floating-point values, but got 'tensor<?x?xi32>'
```

Reply:
Moreover, the integer divide requires additional inputs (e.g. rounding modes), and targets a separate `torch.aten` op to lower from (`Torch_AtenDivTensorMode`). So I'm leaning towards separate ops for cleaner separation of concerns. Will also defer the integer divide case for later, and revise this PR to include `tcp.divf` alone.